src/OrchardCore.Modules/OrchardCore.OpenId/Settings/OpenIdServerSettings.cs

```diff
@@ -19,11 +18,20 @@ public class OpenIdServerSettings
         public PathString UserinfoEndpointPath { get; set; }
         public ISet<string> GrantTypes { get; } = new HashSet<string>(StringComparer.Ordinal);
         public bool UseRollingTokens { get; set; }
+        public IList<SigningKey> SigningKeys { get; } = new List<SigningKey>();
 
         public enum TokenFormat
         {
             Encrypted = 0,
             JWT = 1
         }
+
+        public class SigningKey
+        {
+            public DateTimeOffset? CreationDate { get; set; }
```

Comment:
If this is something that can be part of a recipe, I'd prefer to use `DateTime` and call them as `Utc` (like in content items).

Reply:
Theoretically, it could be part of a recipe, but the signing key payload is supposed to be encrypted using ASP.NET Core Data Protection, which makes it quite hard to register manually.

While one could register a signing key using a recipe, I'd expect the certificate option to be much more popular. This feature is primarily used by the OpenID module, that will create a key and persist it when no certificate was set.